lib/logger/lib/logger/translator.ex

```diff
@@ -29,9 +29,16 @@ defmodule Logger.Translator do
   and the default messages translated by Logger.
   """
 
+  @type result ::
+          {:ok, iodata, keyword}
+          | {:ok, iodata}
+          | :skip
+          | :none
+
   @doc """
   Built-in translation function.
   """
+  @spec translate(Logger.level(), Logger.level(), :format | :report, :logger.report()) :: result
```

Comment:
Probably worth inlining the result here so we don't need to jump around?

Reply:
That would mean any external module that wants to implement its own `transate/4` must copy and paste the result because there isn't a public type (or a behaviour).

Reply:
I think that's alright because you often implement a subset anyway?

Reply:
We can also make this a behaviour, we already have a module anyway. :)

Reply:
> I think that's alright because you often implement a subset anyway?

I could be wrong, but I assumed we had to support every possible `format` or `report`, or purposefully ignore it.  That boils down to doing something like this in a custom translator:

```elixir
def translate(_min, _level, :report, {:logger, message}) do
  # do mappy things
end

def translate(min, level, kind, message) do
  Logger.Translator.translate(min, level, kind, message)
end
```

Which may then return whatever the default `translate/4` does.

> We can also make this a behaviour, we already have a module anyway.

That's fine by me, just give me a 👍 and I'll switch this to a behaviour.

Reply:
We can have several translators. You only handle what you want and return skip for the rest.

but let’s try the behavior route!

Reply:
Slight correction, I believe it should return `:none`, based on this very clear description in the logs:

>  * `:none` - if there is no translation, which triggers the next translator